modules/nf-core/gatk4/haplotypecaller/main.nf

```diff
@@ -29,6 +30,7 @@ process GATK4_HAPLOTYPECALLER {
     def dbsnp_command = dbsnp ? "--dbsnp $dbsnp" : ""
     def interval_command = intervals ? "--intervals $intervals" : ""
     def dragstr_command = dragstr_model ? "--dragstr-params-path $dragstr_model" : ""
+    def bamout_command = args.contains("--bam-writer-type") ? "--bam-output ${prefix.replaceAll('.g\\s*$', '')}.realigned.bam" : ""
```

Comment:
do you want to navigate it with a param or make it so hidden that it can only be done with a custom.config

Reply:
no I figured, that anyone adding this `bam-writer-type` args to Haplotypecaller would actually want to use the `bam-output` one as well

Reply:
ok but this way can only do this via `task.ext.args` not with `params.publish_hc_bams`, is that as you want it?

Reply:
Yes, but I'm thinking that this is a highly specific thing, so for now just a `task.ext.args`, and we'll see about params if this feature could be more useful